### PR TITLE
[Serve] [Doc] Add version warning

### PR DESCRIPTION
**Pull request by architkulkarni**
## Why are these changes needed?

Users often land on the master docs and run into errors running sample code because they are not running the Ray nightly.  This PR changes the Serve front page Starlette migration note from a note into a warning and adds a link to the v:latest version of the docs.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by edoakes — COMMENTED** [review]
Thanks for doing this @architkulkarni. Can we make this even more prominent? e.g., put it at the very top and say a more concise version of: "You're most likely running on the latest release, so some documentation might not line up because we're still changing things quickly, ..."

**Comment by architkulkarni** [comment]
Sure, makes sense.  By the way, do you know if there's a way to make the
warning only show up on the master documentation?  Otherwise, after the
next release (1.3 I guess) the warning will start to appear in the "latest"
documentation, which might be confusing.   There's a Sphinx command in
conf.py that can do something like this for the whole docs (we used to have
it enabled), but I don't know if it's possible to do it just on the front
page of Serve.

**Comment by architkulkarni** [comment]
I guess it's not so bad if this warning persists in the 1.3.0 release, the text still makes logical sense.  We can return to this before the 1.3.0 release and decide if we want to remove the warning then.

**Comment by edoakes** [comment]
@architkulkarni I believe the only way to have that happen would be to cherry-pick a commit into the release branch that removes it. Can definitely do that if it improves things.

**Comment by edoakes** [comment]
Looks like lint is failing:
```
sphinx.errors.SphinxWarning: source/ray-overview/basics.rst:19:undefined label: rayserve (if the link has no caption the label must precede a section header)
Warning, treated as error:
source/ray-overview/basics.rst:19:undefined label: rayserve (if the link has no caption the label must precede a section header)
```

File: doc/source/serve/index.rst [hunk 1]
```diff
@@ -33,8 +37,9 @@ Ray Serve can be used in two primary ways to deploy your models at scale:
   Chat with Ray Serve users and developers on our `forum <https://discuss.ray.io/>`_!
 
 .. note::
-  Starting with Ray version 1.2.0, Ray Serve backends take in a Starlette Request object instead of a Flask Request object.
+  Starting with Ray version 1.2.0, Ray Serve backends take in a Starlette Request object instead of a Flask Request object.
```
Comment (simon-mo) [inline doc/source/serve/index.rst:40]: whitespace

**Review by simon-mo — APPROVED** [review]

**Pull request merged**